coalib/io/Memoize.py

```diff
@@ -0,0 +1,31 @@
+import functools
+
+
+class Memoize:
```

Comment:
Since it's now also turning your functions into properties, you should call it somehow `*_property`, e.g. `cached_property`, `memoized_property`, ... Exact name is up to :P